### ISSUE TITLE
php 8 error in log. setup works and login works cannot add or edit mailbox

### ISSUE THREAD

**Issue by glenb77** [postfixadmin/postfixadmin#574]
On arch linux using arch repo package. had this working previously, but have not checked in some time over several updates. after spending a long time i have the login loading correctly, however when i go to add a mailbox or edit a mailbox, i get a http 500 error and the log for httpd errors shows this:

In bool: expected config forgotten_user_password_reset to be a string, but received a NULL in /usr/share/webapps/postfixadmin/model/Config.php on line 144, referer: http://glenb.net/postfixadmin/list-virtual.php?domain=glenb.net

the setup page indicates the database is current and shows 1844/0 there.

i have $CONF['password_expiration'] = false; in local, i found that recommendation in an earlier issue. the setup page shows only 3 warnings nothing to indicate it should not run. 

i needed to alter save.session in php.ini to a writeable location to get this far. can't figure out why going in to add or edit a mailbox is not working. thanks for any tips

**Comment by DavidGoodwin** [comment]
add `$CONF['forgotten_user_password_reset'] = false;` to your config. 

The error message you highlighted above should just be a warning - and not the cause of the http 500 error. 
I suspect something else is going on.

Do you have a PHP error log configured?

**Comment by glenb77** [comment]
I have just now set a log file in php.ini. I did add the forgotten_user_password as above but still get the 500 error on going to create a mailbox. I commented out lines 144-151 in /mode/Config.php and it will work but displays this above the area to fill out the add mailbox form: 

$CONF['forgotten_user_password_reset'] has an invalid value, should be 'YES' or 'NO'

despite the fact i have the directive in the local php config file.

**Comment by DavidGoodwin** [comment]
I don't think this is a PHP 8 problem - I'm running PHP 8 (fpm) locally and 'master' works fine.

What version of PostfixAdmin are you running?

**Comment by glenb77** [comment]
I agree I'm guessing it's some type of permissions thing I'll keep working on it and get back to you thanks. Blocking that section of code is just to work around most likely because of some permissions error I will try to download from the get tonight and start over thanksOn Dec 15, 2021 1:16 PM, David Goodwin ***@***.***> wrote:
I don't think this is a PHP 8 problem - I'm running PHP 8 (fpm) locally and 'master' works fine.
What version of PostfixAdmin are you running?

**Comment by DavidGoodwin** [comment]
Thanks for clarifying - glad it's all fixed now :)
